components/home/list/LaunchesList.dataLayer.js

```diff
@@ -0,0 +1,52 @@
+import { useQuery, gql } from '@apollo/client';
+import {
+    View,
+    Text,
+  } from "react-native";
+import LaunchesList from './LaunchesList';
+
+const GET_LAUNCHES = gql`
+  query Launches {
+    launches {
+      id
+      details
+      mission_name
+    }
+  }
+`;
+
+function LaunchesListDataLayer() {
+    const { loading, error, data } = useQuery(GET_LAUNCHES);
+
+    if (loading) {
+        return(
+          // TODO: Spinner
+            <View>
+                <Text>
+                    Loading...
+                </Text>
+            </View>
+        )
+    } else if (error) {
```

Comment:
no need for the else if since your previous if statement is a return. If it hit that case it wouldn't be possible to proceed, so you can instead switch to 3 regular if statements on this component